packages/utilities/exposure-scanning/scan/index.js

```diff
@@ -0,0 +1,77 @@
+const replacementStr = '*';
+
+export const scanText = (text, regex) => {
+  let foundTextMatch = false;
+  const scannedText = text.replace(regex, replacementStr);
+
+  if (scannedText !== text) {
+    foundTextMatch = true;
+  }
+
+  return { body: scannedText, foundTextMatch };
+};
+
+export const scanComments = (comments, regex) => {
+  let foundCommentMatch = false;
+  const scannedComments = [];
+
+  comments.forEach(({ id, body }) => {
+    if (regex.test(body)) {
+      foundCommentMatch = true;
+      scannedComments.push({
+        id,
+        body: body.replace(regex, replacementStr),
+      });
+    }
+  });
+
+  return { comments: scannedComments, foundCommentMatch };
+};
```

Comment:
2 ways you can avoid mutable variables and an array push

One with a map and a filter
```js
export const scanComments = (comments, regex) => {
  const scannedComments = comments
    .map(({ id, body }) => {
      if (regex.test(body)) {
        return {
          id,
          body: body.replace(regex, replacementStr),
        };
      }
    })
    .filter(Boolean);
  const foundCommentMatch = scannedComments.length > 0;

  return { comments: scannedComments, foundCommentMatch };
};
```

or using a reduce

```js
export const scanComments = (comments, regex) => {
  const scannedComments = comments.reduce((acc, { id, body }) => {
    if (regex.test(body)) {
      return [
        ...acc,
        {
          id,
          body: body.replace(regex, replacementStr),
        },
      ];
    }

    return acc;
  });
  const foundCommentMatch = scannedComments.length > 0;

  return { comments: scannedComments, foundCommentMatch };
};
```